crates/bevy_ecs/src/archetype.rs

```diff
@@ -109,19 +109,20 @@ impl ArchetypeId {
     }
 }
 
-/// Used in [`AddBundle`] to track whether components in the bundle are newly
+/// Used in [`ArchetypeAfterBundleInsert`] to track whether components in the bundle are newly
 /// added or already existed in the entity's archetype.
 #[derive(Copy, Clone, Eq, PartialEq)]
 pub(crate) enum ComponentStatus {
     Added,
     Existing,
 }
 
-pub(crate) struct AddBundle {
-    /// The target archetype after the bundle is added to the source archetype
+/// Used in [`Edges`] to cache the result of inserting a bundle into the source archetype.
+pub(crate) struct ArchetypeAfterBundleInsert {
```

Comment:
nit: it might be better to name this and other methods `ArchetypeAfterInsertBundle` and `*_insert_bundle()` to match the `insert_bundle` ordering.

Reply:
I went back and forth on this, but I stuck with `bundle_insert` because I thought `insert_bundle` might sound too much like an "action", like `get_archetype_after_insert_bundle` would actually be doing the insert itself. Not married to it though

Reply:
I don't really have too strong feelings here either and I don't think it's worth thinking about too hard. The new names are better, so lets just ship it.